modules/reindex/src/main/java/org/elasticsearch/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -191,6 +200,17 @@
         }
         sourceBuilder.version(needsSourceDocumentVersions);
         sourceBuilder.seqNoAndPrimaryTerm(needsSourceDocumentSeqNoAndPrimaryTerm);
+
+        /*
+         * Do not open scroll if max docs <= scroll size and not resuming on version conflicts
+         */
+        if (mainRequest.getMaxDocs() != -1
+            && mainRequest.getMaxDocs() <= mainRequest.getSearchRequest().source().size()
+            && mainRequest.isAbortOnVersionConflict()) {
+            mainRequest.getSearchRequest().scroll((Scroll) null);
```

Comment:
I believe this main request is the original request created by the client. For the REST client, this is likely fine. But for internal clients, we could risk someone getting a funny effect out of reusing the search request for multiple operations. I am not sure it is a case, but it always annoyed me with the other things we set above.

Could we perhaps instead do this in `buildScrollableResultSource` instead and then create a new `SearchRequest` as a copy of the original when we need this to protect against this unexpected behavior?